### ISSUE TITLE
sqs.Client.TagQueue returns an error but still succeessfully tags a resource

### ISSUE THREAD

**Issue by artyom** [aws/aws-sdk-go-v2#1047]
Confirm by changing [ ] to [x] below to ensure that it's a bug:

- [x] I've gone though the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**

Attempts to tag an SQS queue with [sqs.Client.TagQueue](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/sqs@v0.31.0?#Client.TagQueue) call fails with the following error:

	operation error SQS: TagQueue, https response error StatusCode: 200, RequestID: 551b5c11-ecbd-55de-8d46-046a7bfc9c2d, deserialization failed, failed to decode response body, TagQueueResult node not found

API call actually succeeds, as provided tags appear on a resource.

**Version of AWS SDK for Go?**

0.31.0

**Version of Go (`go version`)?**

go version go1.16beta1 darwin/arm64

**To Reproduce (observed behavior)**

Steps to reproduce the behavior (please share code or minimal repo)

go.mod contents:

```
module sdk-bug

go 1.16

require (
	github.com/aws/aws-sdk-go-v2 v0.31.0
	github.com/aws/aws-sdk-go-v2/config v0.4.0
	github.com/aws/aws-sdk-go-v2/service/sqs v0.31.0
)
```

main.go contents:

```go
package main

import (
	"context"
	"errors"
	"flag"
	"log"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/service/sqs"
)

func main() {
	log.SetFlags(0)
	flag.Parse()
	if err := run(context.Background(), flag.Arg(0)); err != nil {
		log.Fatal(err)
	}
}

func run(ctx context.Context, queueURL string) error {
	if queueURL == "" {
		return errors.New("provide a queue url to tag")
	}
	log.Println("SDK version:", aws.SDKVersion)
	cfg, err := config.LoadDefaultConfig(ctx)
	if err != nil {
		return err
	}
	svc := sqs.NewFromConfig(cfg)
	_, err = svc.TagQueue(ctx, &sqs.TagQueueInput{
		QueueUrl: &queueURL,
		Tags:     map[string]string{"Environment": "staging"},
	})
	return err
}
```

Build and run this program providing an SQS queue URL as a first positional argument.

**Expected behavior**

Program prints SDK version on stderr and finishes with 0 exit status. Resource is tagged.

**Actual behavior**

Program prints SDK version on stderr and finishes with non-zero exit status, printing an error:

    operation error SQS: TagQueue, https response error StatusCode: 200, RequestID: 551b5c11-ecbd-55de-8d46-046a7bfc9c2d, deserialization failed, failed to decode response body, TagQueueResult node not found

Resource *is* tagged.

**Additional context**

Similar error happens with [rds.Client.AddTagsToResource](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/rds@v0.31.0#Client.AddTagsToResource) calls.

**Comment by jasdel** [comment]
Thanks for reporting this issue. We're investigating this bug, and working on a solution. It looks like the SDK is attempting to deserialize a response when it shouldn't be. Will update this GitHub issue when a PR is created to fix this bug.

This issue will also be impacting other API operations where there is not response from the service, but the SDK is trying to unmarshal one. Such as Amazon CloudWatch's PutMetricData.

**Comment by Chandrian** [comment]
Getting the exact same issue with sqs.DeleteMessage call, here's some logs:

SDK 2021/01/15 21:30:36 DEBUG request failed with unretryable error https response error StatusCode: 200, RequestID: xxx, deserialization failed, failed to decode response body, DeleteMessageResult node not found

But it looks like the sqs message was properly deleted.

**Comment by jasdel** [comment]
Thanks for reporting these issues, We've merged in the fix for this bug and will be included in the SDK's next published release.

In the meantime you can use the HEAD of the SDK's repo using go get commands to the git hash, `e18d7141fc06`  You'll need to do this for all SDK modules that you are using in your application.

```bash
# Update SDK's core modules:
go get github.com/aws/smithy-go@fd031a74deeb
go get github.com/aws/aws-sdk-go-v2@e18d7141fc06
go get github.com/aws/aws-sdk-go-v2/service/sts@e18d7141fc06
go get github.com/aws/aws-sdk-go-v2/feature/ec2/imds@e18d7141fc06
go get github.com/aws/aws-sdk-go-v2/credentials@e18d7141fc06
go get github.com/aws/aws-sdk-go-v2/config@e18d7141fc06

# Update SDK API clients your application uses.
go get github.com/aws/aws-sdk-go-v2/service/s3@e18d7141fc06
go get github.com/aws/aws-sdk-go-v2/service/sqs@e18d7141fc06
```

If you get errors about not available try running the above commands with the `GOSUMDB` environment variable disabled.

```bash
export GOSUMDB=off
```

**Comment by jasdel** [comment]
Thanks for reporting these issues. The bug has been fixed, and is included in the SDK's `v1.0.0` release. The SDK is now release for General Availability (GA), and ready for use in production applications.
